### PR TITLE
Fix/handle-all-security-related-issues-of-subscribe-to-plugin

### PR DESCRIPTION
**Pull request by MdAsifHossainNadim**
### Make compatible with wp v6.1.1 & solve security related issues.

In this PR we try to solve form verification related issues & also added capability checking for users. We also try to update coding standard & logical approach.

### CODE REVIEW

**Review by sapayth — CHANGES_REQUESTED** [review]

File: admin/settings.php [hunk 1]
```diff
@@ -54,12 +60,12 @@
 					}
 				}
 			} elseif ( in_array( $key, array( 'notification_subject', 'mailtext', 'confirm_subject', 'confirm_email', 'remind_subject', 'remind_email' ), true ) && ! empty( $_POST[ $key ] ) ) {
-				// email subject and body templates
+				// Email subject and body templates.
 				$this->subscribe2_options[ $key ] = trim( $_POST[ $key ] );
```
[the remark this answers is not in the document]
Comment (MdAsifHossainNadim) [reply admin/settings.php:64]: Is it `wp_sanitize_key` or `sanitize_key`...???

[the remark this answers is not in the document]
Comment (sapayth) [reply admin/settings.php:64]: `sanitize_key`, sorry for the mistake

**Comment by Rat01047** [comment]
Issue-1:
Email Send is not working

Issue-2:
When a `Upload File` is chosen > Click Add More Files > The same file is already selected 

Issue-3:
When you select Add More for > Upload File - There is no way to remove the Add more option

Video for reference:
https://www.awesomescreenshot.com/video/13240279?key=8d604af8ab0b05b559737dbb9ef81ca5

File: traits/ShortcodeTrait.php [hunk 2]
```diff
@@ -190,75 +201,94 @@ public function widget_shortcode( $atts ) {
 				$this->ip = $_POST['ip'];
 				if ( is_int( $this->lockout ) && $this->lockout > 0 ) {
 					$date = gmdate( 'H:i:s.u', $this->lockout );
```
Comment (sapayth) [inline traits/ShortcodeTrait.php:203]: should we use `current_datetime()` instead of `gmdate()`? please reconsider.

**Review by sapayth — CHANGES_REQUESTED** [review]

**Review by sapayth — APPROVED** [review]

**Pull request merged**